mcc/PrintLineWithStyles.c

```diff
@@ -286,55 +299,55 @@ LONG PrintLine(struct InstData *data, LONG x, struct line_node *line, LONG line_
-        {
-          LONG right = MIN(_mright(data->object), xoffset+flow+blockwidth-1);
+        {               
+          LONG left  = MIN(MAX(xoffset, dleft), dright);
+          LONG right = MIN(dright, xoffset+flow+blockwidth-1);
 
           SetAPen(rp, color);
-          RectFill(rp, xoffset, starty, right, starty+data->fontheight-1);
+          if (left < dright || right > dleft)
+            RectFill(rp, left, starty, right, starty+data->fontheight-1);
         }
         else
-        {
-          LONG right = MIN(_mright(data->object), xoffset+flow+blockstart+blockwidth-1);
-
+        {              
+          LONG left  = MIN(MAX(xoffset+flow+blockstart, dleft), dright);
+          LONG right = MIN(dright, xoffset+flow+blockstart+blockwidth-1);
+                       
           SetAPen(rp, cursor ? MUIPEN(data->cursorcolor) : color);
-          RectFill(rp, xoffset+flow+blockstart, starty, right, starty+data->fontheight-1);
-
-          // if the gadget is in inactive state we just draw a skeleton cursor instead
-          if(cursor == TRUE &&
-             isFlagClear(data->flags, FLG_Active) &&
-             isFlagClear(data->flags, FLG_Activated))
+          if (left < dright || right > dleft)
           {
-            DoMethod(data->object, MUIM_DrawBackground, xoffset+flow+blockstart+1, starty+1,
-                                                        blockwidth-2, data->fontheight-2,
-                                                        isFlagSet(data->flags, FLG_InVGrp) ? 0 : _mleft(data->object),
-                                                        isFlagSet(data->flags, FLG_InVGrp) ? data->fontheight*(data->visual_y+line_nr-2) : _mtop(data->object)+data->fontheight * (data->visual_y+line_nr-2),
-                                                        0);
+            RectFill(rp, left, starty, right, starty+data->fontheight-1);
+
+            // if the gadget is in inactive state we just draw a skeleton cursor instead
+            if(cursor == TRUE &&
+               isFlagClear(data->flags, FLG_Active) &&
+               isFlagClear(data->flags, FLG_Activated))
+            {
+              DoMethod(data->object, MUIM_DrawBackground, left+1, starty+1,
+                                                          blockwidth-2, data->fontheight-2,
+                                                          isFlagSet(data->flags, FLG_InVGrp) ? 0 : _mleft(data->object),
+                                                          isFlagSet(data->flags, FLG_InVGrp) ? data->fontheight*(data->visual_y+line_nr-2) : _mtop(data->object)+data->fontheight * (data->visual_y+line_nr-2),
+                                                          0);
+            }
           }
         }
       }
 
+                       
+      {                  
+        LONG  x_start = MAX(dleft, (blockwidth ? xoffset+blockstart+blockwidth+flow : xoffset+blockstart+blockwidth)),
```

Comment:
now that you touch these lines please add separate variable definition on each line and don't concatenate them via ','.